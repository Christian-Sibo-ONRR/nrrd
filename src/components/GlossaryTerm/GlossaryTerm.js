import React from 'react'
import PropTypes from 'prop-types'

import { useStaticQuery, graphql } from 'gatsby'

import { IconQuestionCircleImg } from '../images'

import makeStyles from '@material-ui/styles/makeStyles'
import useTheme from '@material-ui/styles/useTheme'
import Tooltip from '@material-ui/core/Tooltip'

const useStyles = makeStyles(theme => (
  {
    tooltip: {
      backgroundColor: theme.palette.info.dark,
      color: theme.palette.common.white,
      boxShadow: theme.shadows[2],
      fontSize: theme.typography.fontSize,
    },
    arrow: {
      color: theme.palette.info.dark,
    },
    term: {
      borderBottom: '1px dotted',
      cursor: 'pointer'
    }
  }
))

const GlossaryTerm = ({ termKey, children, ...rest }) => {
  const theme = useTheme()
  const styles = useStyles(theme)
  const results = useStaticQuery(graphql`
    query GlossaryTermsQuery {
      allMdx(filter: {fileAbsolutePath: {regex: "/content-partials/Glossary/"}, mdxAST: {}}) {
        nodes {
          frontmatter {
            terms {
              definition
              name
            }
          }
        }
      }
    }
  `)

  const terms = results.allMdx.nodes[0].frontmatter.terms
  const termId = termKey || children

  const termResults = terms.filter(term => termId.toLowerCase() === term.name.toLowerCase())

  if (termResults.length > 1) {
    console.warning(`Found more than 1 definition for the termId: ${ termId }. Will use the first result returned.`, termResults)
  }

  if (termResults.length === 0) {
    throw new Error(`Found no definitions for the termId: ${ termId }.`)
  }

  const TermDisplay = React.forwardRef((props, ref) => (
    <span {...props} ref={ref}>
      {`${ children } `}<IconQuestionCircleImg />
    </span>
  ))

  return (
    <Tooltip title={termResults[0].definition} classes={styles} leaveDelay={200} arrow placement="top">
      <TermDisplay className={styles.term} {...rest} />
    </Tooltip>

  )
}

GlossaryTerm.propTypes = {
  /** If the key for looking up the glossary term is differnt then the children property use this to override */
  termKey: PropTypes.string,
}

export default GlossaryTerm
