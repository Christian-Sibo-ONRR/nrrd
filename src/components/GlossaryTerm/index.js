export { default as GlossaryTerm } from './GlossaryTerm'
