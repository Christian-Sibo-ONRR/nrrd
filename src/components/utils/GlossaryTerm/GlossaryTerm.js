import React, { Fragment, useContext } from 'react'
import { makeStyles } from '@material-ui/core/styles'
import { StoreContext } from '../../../store'

import GlossaryIcon from '-!svg-react-loader!../../../img/svg/icon-question-circle.svg'

// GlossaryTerm Styles
const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  glossaryTerm: {
    borderBottom: `1px dotted ${ theme.palette.common.black }`,
    cursor: 'pointer',
    '&:hover': {
      borderBottom: `1px solid ${ theme.palette.common.black }`,
    }
  },
  iconQuestion: {
    width: '16px',
    height: '16px',
    marginLeft: theme.spacing(0.2)
  }
}))

const GlossaryTerm = ({ termKey, children }) => {
  const classes = useStyles()
  const { setGlossaryTermSelected } = useContext(StoreContext)

  const getTerm = () => {
    if (termKey) {
      return termKey
    }
    else if (Array.isArray(children)) {
      return children[0]
    }
    return children
  }

  return (
    <Fragment>
      <span className={classes.glossaryTerm} title='Click to define' tabIndex='0'
        onClick={() => setGlossaryTermSelected(getTerm())}>
        {children} <GlossaryIcon className={classes.iconQuestion} />
      </span>
    </Fragment>
  )
}

export default GlossaryTerm
