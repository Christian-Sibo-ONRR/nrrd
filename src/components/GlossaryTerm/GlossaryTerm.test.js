/* eslint-disable no-undef */
import React from 'react'
import GlossaryTerm from './GlossaryTerm'
import { create } from 'react-test-renderer'
import theme from '../../js/mui/theme'
import { ThemeProvider } from '@material-ui/core/styles'

describe('Glossary Term:', () => {
  function MockedTheme ({ children }) {
    return (
      <ThemeProvider theme={theme}>
        {children}
      </ThemeProvider>
    )
  }

  test('Match snapshot', () => expect(create(<MockedTheme><GlossaryTerm>EITI Standard</GlossaryTerm></MockedTheme>).toJSON()).toMatchSnapshot())
})
