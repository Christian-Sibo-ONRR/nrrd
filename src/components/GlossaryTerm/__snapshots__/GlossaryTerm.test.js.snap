// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Glossary Term: Match snapshot 1`] = `
<span
  aria-describedby={null}
  className="makeStyles-term-3"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  title="The Extractive Industries Transparency Initiative Standard is an international standard for openness around the management of revenue from natural resources. Governments disclose how much they receive from extractive companies operating in their country and these companies disclose how much they pay. Governments sign up to implement the EITI Standard and must meet seven requirements. In 2017, the U.S. withdrew from EITI as an Implementing Country, but remains committed to institutionalizing the EITI principles of transparency and accountability."
>
  EITI Standard 
  <test-file-stub
    style={
      Object {
        "fill": "currentColor",
        "height": "16px",
        "marginLeft": "1.6px",
        "width": "16px",
      }
    }
  />
</span>
`;
